account_invoice_change_currency/hooks.py

```diff
@@ -0,0 +1,29 @@
+import logging
+
+from odoo import fields, tools
+
+_logger = logging.getLogger(__name__)
+
+
+def pre_init_hook(cr):
+    """
+    Initializing column custom_rate on table account_move
+    for the improvement in performance to avoid long
+    duration in databases with thousand of moves
+    """
+    _logger.info("Initializing column custom_rate on table account_move")
+    tools.create_column(
+        cr=cr,
+        tablename="account_move",
+        columnname="custom_rate",
+        columntype=fields.Float.type,
```

Comment:
```suggestion
        columntype=fields.Float.column_type[1],,
```

To follow the same way as Odoo [does to create columns](https://github.com/odoo/odoo/blob/da6bd90eed55186e00e330f390c4d0d0b351167c/odoo/fields.py#L841)

Reply:
Take a look what I did, this direct solution was not working for me.